### ISSUE TITLE
 🅿️⛱️  onboard checklist for new projects under org umbrella

### ISSUE THREAD

**Issue by DerekNonGeneric** [OpenINF/.github#723]
i have been meaning to do this for a very long time:

every project repo transferred under @OpenINF umbrella has to undergo a certain procedure before we can properly maintain it or bring it up to a recovered state where we may begin to have to in the context of our project lifecycle(s); let us flesh that out (minimally at first, then more-so in the short-term)

**Comment by DerekNonGeneric** [comment]
🤖  https://probot.github.io/apps/settings/

the screenshot above may or not be close to what we normally expect well-run projects to use

**Comment by DerekNonGeneric** [comment]
<div class="markdown-body">
        <p>This GitHub App syncs repository settings defined in <code class="notranslate">.github/settings.yml</code> to GitHub, enabling Pull Requests for repository settings.</p>
<p><a href="https://vercel.com?utm_source=repository-settings&amp;utm_campaign=oss" rel="nofollow"><img src="https://github.com/repository-settings/app/raw/master/assets/powered-by-vercel.svg" alt="Powered by Vercel" style="max-width: 100%;"></a></p>
<h2>Usage</h2>
<ol>
<li><strong><a href="https://github.com/apps/settings">Install the app</a></strong>.</li>
<li>Create a <code class="notranslate">.github/settings.yml</code> file in your repository. Changes to this file on the default branch will be synced to GitHub.</li>
</ol>
<p>All top-level settings are optional. Some plugins do have required fields.</p>
<div class="highlight highlight-source-yaml"><pre class="notranslate"><span class="pl-c"><span class="pl-c">#</span> These settings are synced to GitHub by https://probot.github.io/apps/settings/</span>

<span class="pl-ent">repository</span>:
  <span class="pl-c"><span class="pl-c">#</span> See https://docs.github.com/en/rest/reference/repos#update-a-repository for all available settings.</span>

  <span class="pl-c"><span class="pl-c">#</span> The name of the repository. Changing this will rename the repository</span>
  <span class="pl-ent">name</span>: <span class="pl-s">repo-name</span>

  <span class="pl-c"><span class="pl-c">#</span> A short description of the repository that will show up on GitHub</span>
  <span class="pl-ent">description</span>: <span class="pl-s">description of repo</span>

  <span class="pl-c"><span class="pl-c">#</span> A URL with more information about the repository</span>
  <span class="pl-ent">homepage</span>: <span class="pl-s">https://example.github.io/</span>

  <span class="pl-c"><span class="pl-c">#</span> A comma-separated list of topics to set on the repository</span>
  <span class="pl-ent">topics</span>: <span class="pl-s">github, probot</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to make the repository private, or `false` to make it public.</span>
  <span class="pl-ent">private</span>: <span class="pl-c1">false</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable issues for this repository, `false` to disable them.</span>
  <span class="pl-ent">has_issues</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable projects for this repository, or `false` to disable them.</span>
  <span class="pl-c"><span class="pl-c">#</span> If projects are disabled for the organization, passing `true` will cause an API error.</span>
  <span class="pl-ent">has_projects</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable the wiki for this repository, `false` to disable it.</span>
  <span class="pl-ent">has_wiki</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable downloads for this repository, `false` to disable them.</span>
  <span class="pl-ent">has_downloads</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Updates the default branch for this repository.</span>
  <span class="pl-ent">default_branch</span>: <span class="pl-s">master</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to allow squash-merging pull requests, or `false` to prevent</span>
  <span class="pl-c"><span class="pl-c">#</span> squash-merging.</span>
  <span class="pl-ent">allow_squash_merge</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to allow merging pull requests with a merge commit, or `false`</span>
  <span class="pl-c"><span class="pl-c">#</span> to prevent merging pull requests with merge commits.</span>
  <span class="pl-ent">allow_merge_commit</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to allow rebase-merging pull requests, or `false` to prevent</span>
  <span class="pl-c"><span class="pl-c">#</span> rebase-merging.</span>
  <span class="pl-ent">allow_rebase_merge</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable automatic deletion of branches on merge, or `false` to disable</span>
  <span class="pl-ent">delete_branch_on_merge</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable automated security fixes, or `false` to disable</span>
  <span class="pl-c"><span class="pl-c">#</span> automated security fixes.</span>
  <span class="pl-ent">enable_automated_security_fixes</span>: <span class="pl-c1">true</span>

  <span class="pl-c"><span class="pl-c">#</span> Either `true` to enable vulnerability alerts, or `false` to disable</span>
  <span class="pl-c"><span class="pl-c">#</span> vulnerability alerts.</span>
  <span class="pl-ent">enable_vulnerability_alerts</span>: <span class="pl-c1">true</span>

<span class="pl-c"><span class="pl-c">#</span> Labels: define labels for Issues and Pull Requests</span>
<span class="pl-ent">labels</span>:
  - <span class="pl-ent">name</span>: <span class="pl-s">bug</span>
    <span class="pl-ent">color</span>: <span class="pl-s">CC0000</span>
    <span class="pl-ent">description</span>: <span class="pl-s">An issue with the system 🐛.</span>

  - <span class="pl-ent">name</span>: <span class="pl-s">feature</span>
    <span class="pl-c"><span class="pl-c">#</span> If including a `#`, make sure to wrap it with quotes!</span>
    <span class="pl-ent">color</span>: <span class="pl-s"><span class="pl-pds">'</span>#336699<span class="pl-pds">'</span></span>
    <span class="pl-ent">description</span>: <span class="pl-s">New functionality.</span>

  - <span class="pl-ent">name</span>: <span class="pl-s">Help Wanted</span>
    <span class="pl-c"><span class="pl-c">#</span> Provide a new name to rename an existing label</span>
    <span class="pl-ent">new_name</span>: <span class="pl-s">first-timers-only</span>

<span class="pl-c"><span class="pl-c">#</span> Milestones: define milestones for Issues and Pull Requests</span>
<span class="pl-ent">milestones</span>:
  - <span class="pl-ent">title</span>: <span class="pl-s">milestone-title</span>
    <span class="pl-ent">description</span>: <span class="pl-s">milestone-description</span>
    <span class="pl-c"><span class="pl-c">#</span> The state of the milestone. Either `open` or `closed`</span>
    <span class="pl-ent">state</span>: <span class="pl-s">open</span>

<span class="pl-c"><span class="pl-c">#</span> Collaborators: give specific users access to this repository.</span>
<span class="pl-c"><span class="pl-c">#</span> See https://docs.github.com/en/rest/reference/repos#add-a-repository-collaborator for available options</span>
<span class="pl-ent">collaborators</span>:
  <span class="pl-c"><span class="pl-c">#</span> - username: bkeepers</span>
  <span class="pl-c"><span class="pl-c">#</span>   permission: push</span>
  <span class="pl-c"><span class="pl-c">#</span> - username: hubot</span>
  <span class="pl-c"><span class="pl-c">#</span>   permission: pull</span>

  <span class="pl-c"><span class="pl-c">#</span> Note: `permission` is only valid on organization-owned repositories.</span>
  <span class="pl-c"><span class="pl-c">#</span> The permission to grant the collaborator. Can be one of:</span>
  <span class="pl-c"><span class="pl-c">#</span> * `pull` - can pull, but not push to or administer this repository.</span>
  <span class="pl-c"><span class="pl-c">#</span> * `push` - can pull and push, but not administer this repository.</span>
  <span class="pl-c"><span class="pl-c">#</span> * `admin` - can pull, push and administer this repository.</span>
  <span class="pl-c"><span class="pl-c">#</span> * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.</span>
  <span class="pl-c"><span class="pl-c">#</span> * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.</span>

<span class="pl-c"><span class="pl-c">#</span> See https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions for available options</span>
<span class="pl-ent">teams</span>:
  - <span class="pl-ent">name</span>: <span class="pl-s">core</span>
    <span class="pl-c"><span class="pl-c">#</span> The permission to grant the team. Can be one of:</span>
    <span class="pl-c"><span class="pl-c">#</span> * `pull` - can pull, but not push to or administer this repository.</span>
    <span class="pl-c"><span class="pl-c">#</span> * `push` - can pull and push, but not administer this repository.</span>
    <span class="pl-c"><span class="pl-c">#</span> * `admin` - can pull, push and administer this repository.</span>
    <span class="pl-c"><span class="pl-c">#</span> * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.</span>
    <span class="pl-c"><span class="pl-c">#</span> * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.</span>
    <span class="pl-ent">permission</span>: <span class="pl-s">admin</span>
  - <span class="pl-ent">name</span>: <span class="pl-s">docs</span>
    <span class="pl-ent">permission</span>: <span class="pl-s">push</span>

<span class="pl-ent">branches</span>:
  - <span class="pl-ent">name</span>: <span class="pl-s">master</span>
    <span class="pl-c"><span class="pl-c">#</span> https://docs.github.com/en/rest/reference/repos#update-branch-protection</span>
    <span class="pl-c"><span class="pl-c">#</span> Branch Protection settings. Set to null to disable</span>
    <span class="pl-ent">protection</span>:
      <span class="pl-c"><span class="pl-c">#</span> Required. Require at least one approving review on a pull request, before merging. Set to null to disable.</span>
      <span class="pl-ent">required_pull_request_reviews</span>:
        <span class="pl-c"><span class="pl-c">#</span> The number of approvals required. (1-6)</span>
        <span class="pl-ent">required_approving_review_count</span>: <span class="pl-c1">1</span>
        <span class="pl-c"><span class="pl-c">#</span> Dismiss approved reviews automatically when a new commit is pushed.</span>
        <span class="pl-ent">dismiss_stale_reviews</span>: <span class="pl-c1">true</span>
        <span class="pl-c"><span class="pl-c">#</span> Blocks merge until code owners have reviewed.</span>
        <span class="pl-ent">require_code_owner_reviews</span>: <span class="pl-c1">true</span>
        <span class="pl-c"><span class="pl-c">#</span> Specify which users and teams can dismiss pull request reviews. Pass an empty dismissal_restrictions object to disable. User and team dismissal_restrictions are only available for organization-owned repositories. Omit this parameter for personal repositories.</span>
        <span class="pl-ent">dismissal_restrictions</span>:
          <span class="pl-ent">users</span>: <span class="pl-s">[]</span>
          <span class="pl-ent">teams</span>: <span class="pl-s">[]</span>
      <span class="pl-c"><span class="pl-c">#</span> Required. Require status checks to pass before merging. Set to null to disable</span>
      <span class="pl-ent">required_status_checks</span>:
        <span class="pl-c"><span class="pl-c">#</span> Required. Require branches to be up to date before merging.</span>
        <span class="pl-ent">strict</span>: <span class="pl-c1">true</span>
        <span class="pl-c"><span class="pl-c">#</span> Required. The list of status checks to require in order to merge into this branch</span>
        <span class="pl-ent">contexts</span>: <span class="pl-s">[]</span>
      <span class="pl-c"><span class="pl-c">#</span> Required. Enforce all configured restrictions for administrators. Set to true to enforce required status checks for repository administrators. Set to null to disable.</span>
      <span class="pl-ent">enforce_admins</span>: <span class="pl-c1">true</span>
      <span class="pl-c"><span class="pl-c">#</span> Prevent merge commits from being pushed to matching branches</span>
      <span class="pl-ent">required_linear_history</span>: <span class="pl-c1">true</span>
      <span class="pl-c"><span class="pl-c">#</span> Required. Restrict who can push to this branch. Team and user restrictions are only available for organization-owned repositories. Set to null to disable.</span>
      <span class="pl-ent">restrictions</span>:
        <span class="pl-ent">apps</span>: <span class="pl-s">[]</span>
        <span class="pl-ent">users</span>: <span class="pl-s">[]</span>
        <span class="pl-ent">teams</span>: <span class="pl-s">[]</span></pre></div>
<h3>Notes</h3>
<ol>
<li>Label color can also start with <code class="notranslate">#</code>, e.g. <code class="notranslate">color: '#F341B2'</code>. Make sure to wrap it with quotes!</li>
<li>Each top-level element under branch protection must be filled (eg: <code class="notranslate">required_pull_request_reviews</code>, <code class="notranslate">required_status_checks</code>, <code class="notranslate">enforce_admins</code> and <code class="notranslate">restrictions</code>). If you don't want to use one of them you must set it to <code class="notranslate">null</code> (see comments in the example above). Otherwise, none of the settings will be applied.</li>
</ol>
<h3>Inheritance</h3>
<p>This app is built with <a href="https://github.com/probot/probot">probot</a>, and thus uses the <a href="https://github.com/probot/octokit-plugin-config">octokit-plugin-config</a>. This means you can inherit settings from another repo, and only override what you want to change.</p>
<p>Individual settings in the arrays listed under <code class="notranslate">labels</code>, <code class="notranslate">teams</code> (once it is supported) and <code class="notranslate">branches</code> will be merged with the base repo if the <code class="notranslate">name</code> of an element in the array matches the <code class="notranslate">name</code> of an element in the corresponding array in the base repo. A possible future enhancement would be to make that work for the other settings arrays based on <code class="notranslate">username</code>, or <code class="notranslate">title</code>. This is not currently supported.</p>
<p>To further clarify: Inheritance within the Protected Branches plugin allows you to override specific settings per branch. For example, your <code class="notranslate">.github</code> repo may set default protection on the <code class="notranslate">master</code> branch. You can then include <code class="notranslate">master</code> in your <code class="notranslate">branches</code> array, and only override the <code class="notranslate">required_approving_review_count</code>.<br>
Alternatively, you might only have a branch like <code class="notranslate">develop</code> in your <code class="notranslate">branches</code> array, and would still get <code class="notranslate">master</code> protection from your base repo.</p>
<h2>Security Implications</h2>
<p><strong>WARNING:</strong> Note that this app inherently <em>escalates anyone with <code class="notranslate">push</code> permissions to the <strong>admin</strong> role</em>, since they can push config settings to the <code class="notranslate">master</code> branch, which will be synced. In a future, we may add restrictions to allow changes to the config file to be merged only by specific people/teams, or those with <strong>admin</strong> access <em>(via a combination of protected branches, required statuses, and branch restrictions)</em>. Until then, use caution when merging PRs and adding collaborators.</p>
<p>Until restrictions are added in this app, one way to preserve admin/push permissions is to utilize the <a href="https://help.github.com/articles/about-codeowners/">GitHub CodeOwners feature</a> to set one or more administrative users as the code owner of the <code class="notranslate">.github/settings.yml</code> file, and turn on "require code owner review" for the master branch. This does have the side effect of requiring code owner review for the entire branch, but helps preserve permission levels.</p>

    </div>

**Comment by DerekNonGeneric** [comment]
more apps:

![image](https://github.com/OpenINF/.github/assets/17770407/2e71ed0b-d4ce-4917-ba29-5799665d50ad)

**Comment by openinfbot** [comment]
@DerekNonGeneric, please first do:

> # Labels: define labels for Issues and Pull Requests
> labels:
